Prac05/emails.py

```diff
@@ -0,0 +1,27 @@
+"""
+a program which stores your email with the key being your name
+"""
+
+def main():
+    email_to_name = {}
+    email = input("Email: ")
+    while email != "":
+        name = get_name_from_email(email)
+        confirmation = input("Is your name {}? (Y/n) ".format(name))
+        if confirmation.upper() != "Y" and confirmation != "":
+            name = input("Name: ")
+        email_to_name[email] = name
+        email = input("Email: ")
+
+    for email, name in email_to_name.items():
+        print("{} ({})".format(name, email))
+
+
+def get_name_from_email(email):
+    prefix = email.split('@')[0]
+    parts = prefix.split('.')
```

Comment:
prefix and parts could use the same variable name

Reply:
Why would reusing the same name for different things be a good idea? 🤔